src/java.base/share/classes/java/util/regex/MatchResult.java

```diff
@@ -74,7 +77,44 @@ public interface MatchResult {
      *          If there is no capturing group in the pattern
      *          with the given index
      */
-    public int start(int group);
+    int start(int group);
+
+    /**
+     * Returns the start index of the subsequence captured by the given
+     * <a href="Pattern.html#groupname">named-capturing group</a> during the
+     * previous match operation.
+     *
+     * @param  name
+     *         The name of a named-capturing group in this matcher's pattern
+     *
+     * @return  The index of the first character captured by the group,
+     *          or {@code -1} if the match was successful but the group
+     *          itself did not match anything
+     *
+     * @throws  IllegalStateException
+     *          If no match has yet been attempted,
+     *          or if the previous match operation failed
+     *
+     * @throws  IllegalArgumentException
+     *          If there is no capturing group in the pattern
+     *          with the given name
+     *
+     * @implNote
+     * The default implementation of this method makes use of the map returned
+     * by {@link #namedGroups()}. It is thus sufficient to override
+     * {@link #namedGroups()} for this method to work. However, overriding this
+     * method directly might be preferable for performance or for other reasons.
+     *
+     * @implSpec
+     * The default implementation of this method throws
+     * {@link UnsupportedOperationException} if {@link #namedGroups()} is not
+     * overridden.
```

Comment:
The essential thing for `@implSpec` is to describe "self-use" of methods on this object. This is important for subclassers to know whether they can inherit the default implementation or whether they should override it. It looks like `start(String)` does the following:

- calls namedGroups() to obtain a mapping from group names to group numbers, propagating UOE if namedGroups() throws it
- if `name` is not present in the group map, throws IAE
- calls `start()` on the group number obtained from the map, and returns that value

I don't think we need to go to the level of detail about whether `get` or `containsKey` is called on the map, but I think the self-calls to `namedGroups()` and `start(int)` are important.

Similar comments apply to the `@implSpec` comments of `end(String)` and `group(String)`.